### PR TITLE
Replace name/name_val with id/name in grammar for better semantics

### PR DESCRIPTION
**Pull request by martin-fleck-at**

### CODE REVIEW

File: extensions/crossmodel-lang/src/language-server/generated/ast.ts [hunk 1]
```diff
@@ -8,200 +8,203 @@ import type { AstNode, Reference, ReferenceInfo, TypeMetaData } from 'langium';
...
+export function isIdentifier(item: unknown): item is Identifier {
+    return typeof item === 'string';
 }
 
 export interface CrossModelRoot extends AstNode {
-   readonly $type: 'CrossModelRoot';
-   diagram?: SystemDiagram;
-   entity?: Entity;
-   relationship?: Relationship;
+    readonly $type: 'CrossModelRoot';
```
[the remark this answers is not in the document]
Comment (harmen-xb) [reply extensions/crossmodel-lang/src/language-server/generated/ast.ts:27]: Also it looks like the 'prettier' plugin did some housekeeping here. Does that make sense for generated files?

**Review by harmen-xb — COMMENTED** [review]
Hi Martin,

Thanks for the changes, majority looks very good.

I had some questions/remarks in the comments.

File: extensions/crossmodel-lang/src/language-server/cross-model-naming.ts [hunk 2]
```diff
@@ -2,17 +2,38 @@
...
+}
+
+export interface IdProvider extends NameProvider {
+   getNodeId(node?: AstNode): string | undefined;
+   getLocalId(node?: AstNode): string | undefined;
+   getExternalId(node?: AstNode): string | undefined;
+}
 
 /**
  * A name provider that returns the fully qualified name of a node by default but also exposes methods to get other names:
```
[the remark this answers is not in the document]
Comment (martin-fleck-at) [reply extensions/crossmodel-lang/src/language-server/cross-model-naming.ts:31]: Absolutely correct, I'll update it in a follow-up commit in a second.

File: extensions/crossmodel-lang/src/language-server/cross-model-scope.ts [hunk 3]
```diff
@@ -94,16 +94,16 @@ export class CrossModelScopeComputation extends DefaultScopeComputation {
          description = this.descriptions.createDescription(node, packageQualifiedName, document);
          exports.push(new PackageExternalAstNodeDescription(packageId, packageQualifiedName, description));
       }
-      const documentLocalName = this.nameProvider.getQualifiedName(node);
-      if (documentLocalName && description) {
-         exports.push(new PackageLocalAstNodeDescription(packageId, documentLocalName, description));
+      const packageLocalName = this.idProvider.getLocalId(node);
```
[the remark this answers is not in the document]
Comment (martin-fleck-at) [reply extensions/crossmodel-lang/src/language-server/cross-model-scope.ts:97]: See previous answer ;-) I'll change it to name.

File: extensions/crossmodel-lang/src/language-server/generated/ast.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (harmen-xb) [reply extensions/crossmodel-lang/src/language-server/generated/ast.ts:27]: Ok, just strange it's being mixed here in the generated file then :). But thanks for the explanation.

File: extensions/crossmodel-lang/src/language-server/cross-model-validator.ts [hunk 4]
```diff
@@ -26,27 +24,17 @@ export function registerValidationChecks(services: CrossModelServices): void {
...
-   }
+   constructor(protected services: CrossModelServices) {}
 
-   checkEntityHasNecessaryFields(entity: Entity, accept: ValidationAcceptor): void {
-      if (!entity.name) {
-         accept('error', 'Entity missing id field', { node: entity, property: 'name' });
+   checkUniqueId(node: AstNode, accept: ValidationAcceptor): void {
+      const elementName = this.services.references.IdProvider.getNodeId(node);
+      if (!elementName) {
+         return;
```
[the remark this answers is not in the document]
Comment (harmen-xb) [reply extensions/crossmodel-lang/src/language-server/cross-model-validator.ts:44]: I see you alreay fixed it :)

File: extensions/crossmodel-lang/src/language-server/generated/ast.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sailingKieler) [reply extensions/crossmodel-lang/src/language-server/generated/ast.ts:27]: @martin-fleck-at Thanks a lot for referencing [eclipse-langium/langium] here that made me stumble over this discussion. Actually the referenced change didn't change anything wrt. this, the inconsistent generation has always been there, see e.g. a commit from the early days of Langium contributing one of the example languages:
https://github.com/eclipse-langium/langium/commit/6e927f9d5b8069ed00789b9fdb34b836fb101ddf#diff-f173f2c78506bfc50069545c50a2be09d954fa6eb2bad3a723f09edb46a622c8R23-R25

I filed eclipse-langium/langium#1332 on that.

**Pull request merged**